client/my-sites/controller.js

```diff
@@ -109,7 +109,7 @@ module.exports = {
 	 * Set up site selection based on last URL param and/or handle no-sites error cases
 	 */
 	siteSelection( context, next ) {
-		const siteID = route.getSiteFragment( context.path );
+		const siteID = context.params.site_id;
```

Comment:
Why are you introducing a new url pattern, by the way? Urls in calypso should have the site at the end, the only cases where something goes to the right of a site id is when the resource is from the site itself (a post id, for instance).

In this case `/settings/export/guided/:host_slug/:site_id`.